target/configs.go

```diff
@@ -0,0 +1,143 @@
+package target
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/upsight/ron/color"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Configs is a mapping of filename to target file.
+type Configs struct {
+	Files  []*File
+	Env    *Env // the environment variables used for commands.
+	StdOut io.Writer
+	StdErr io.Writer
+}
+
+// NewConfigs takes a default set of yaml in config format and then
+// overrides them with a new set of config target replacements.
+func NewConfigs(env *Env, configs []*RawConfig, stdOut io.Writer, stdErr io.Writer) (*Configs, error) {
+	if stdOut == nil {
+		stdOut = os.Stdout
+	}
+	if stdErr == nil {
+		stdErr = os.Stderr
+	}
+
+	t := &Configs{
+		Files:  []*File{},
+		Env:    env,
+		StdOut: stdOut,
+		StdErr: stdErr,
+	}
+	for _, config := range configs {
+		var targets map[string]*Target
+		if err := yaml.Unmarshal([]byte(config.Targets), &targets); err != nil {
+			return nil, err
+		}
+		// initialize io for each target.
+		for name, target := range targets {
+			target.W = stdOut
+			target.WErr = stdErr
+			if target.Name == "" {
+				target.Name = name
+			}
+			target.targetConfigs = t
+		}
+
+		f := &File{
+			rawConfig: config,
+			Filepath:  config.Filepath,
+			Targets:   targets,
+		}
+		t.Files = append(t.Files, f)
+	}
+	sort.Sort(t)
+	return t, nil
+}
+
+// List prints out each target and its before and after targets.
+func (tc *Configs) List(verbose bool, fuzzy string) {
+	filePrefix, fuzzyTarget := splitTarget(fuzzy)
+LOOP_FILES:
+	for _, tf := range tc.Files {
+		// If a file prefix is provided check this file matches.
+		if filePrefix != "" && tf.Basename() != filePrefix {
+			continue LOOP_FILES
+		}
+
+		targetNameWidth := 0
+		targetNames := []string{}
+	LOOP_TARGETS:
+		for k := range tf.Targets {
+			if len(k) > targetNameWidth {
+				targetNameWidth = len(k)
+			}
+			if fuzzyTarget != "" {
+				if ok, _ := filepath.Match(fuzzyTarget, k); !ok {
+					continue LOOP_TARGETS
```

Comment:
Seems like you don't need these labels, since you're only continuing the immediately surrounding loops. A bare continue should do the same thing.